tests/subsys/fs/zms/src/main.c

```diff
@@ -4,15 +4,14 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#if defined(CONFIG_BOARD_QEMU_X86) || defined(CONFIG_ARCH_POSIX)
 /*
- * This test is designed to be run using flash-simulator which provide
+ * Enable tests designed to be run using flash-simulator which provide
  * functionality for flash property customization and emulating errors in
  * flash operation in parallel to regular flash API.
  * Test should be run on qemu_x86 or native_sim target.
  */
-
-#if !defined(CONFIG_BOARD_QEMU_X86) && !defined(CONFIG_ARCH_POSIX)
-#error "Run only on qemu_x86 or a posix architecture based target (for ex. native_sim)"
+#define TEST_ZMS_SIMULATOR
```

Comment:
Would it be possible to make this via this test specific Kconfig, making the TEST_ZMS_SIMULATOR as Kconfig that would default to true if CONFIG_BOARD_QEMU_X86 or CONFIG_ARCH_POSIX?
This way it should be possible to set the Kconfig TEST_ZMS_SIMULATOR  via proj.conf or sample.yaml/test.yaml.

Reply:
> Would it be possible to make this via this test specific Kconfig, making the TEST_ZMS_SIMULATOR as Kconfig that would default to true if CONFIG_BOARD_QEMU_X86 or CONFIG_ARCH_POSIX? This way it should be possible to set the Kconfig TEST_ZMS_SIMULATOR via proj.conf or sample.yaml/test.yaml.

OK, I will do it.

Reply:
@de-nordic 
The requested CONFIG_TEST_ZMS_SIMULATOR has been added.